packages/matrix-client-server/src/utils/userInteractiveAuthentication.ts

```diff
@@ -6,29 +6,129 @@ import {
   type AuthenticationData,
   type ClientServerDb,
   type Config,
-  type AppServiceRegistration
+  type AppServiceRegistration,
+  type ThreepidCreds,
+  type AuthenticationFlowContent,
+  type AuthenticationTypes
 } from '../types'
 import { Hash, randomString } from '@twake/crypto'
 import type MatrixDBmodified from '../matrixDb'
 import { errMsg, jsonContent, send, toMatrixId } from '@twake/utils'
 export type UiAuthFunction = (
   req: Request | http.IncomingMessage,
   res: Response | http.ServerResponse,
-  callback: (data: any) => void
+  allowedFlows: AuthenticationFlowContent,
+  callback: (data: any, userId: string | null) => void
 ) => void
 
 interface requestBody {
   auth?: AuthenticationData
   [key: string]: any // others parameters given in request body
 }
 
+const getParams = (type: AuthenticationTypes): any => {
+  // for now only terms has params, spec is unclear about the other types. Add params here if needed in other endpoints
+  switch (type) {
+    case 'm.login.terms':
+      return {
+        policies: {
+          terms_of_service: {
+            version: '1.2',
+            en: {
+              name: 'Terms of Service',
+              url: 'https://example.org/somewhere/terms-1.2-en.html'
+            },
+            fr: {
+              name: "Conditions d'utilisation",
+              url: 'https://example.org/somewhere/terms-1.2-fr.html'
+            }
+          }
+        }
+      }
```

Comment:
Temporary fixed value ?

Reply:
Yes, I believe these should be included in the config in the future but I've just included this from an example of the spec for testing purposes. I'll add a comment to clarify this

Reply:
Then add a TODO

Reply:
Done